src/integrationTest/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/component/testutils/IaCaseHomeOfficeIntegrationApiClient.java

```diff
@@ -0,0 +1,119 @@
+                                       + "kiod3y2G-3BeRq-0otx7gm4-D6UBmJ7vKuct5I9qZsAjYYCb8vMQQAU-sAnD6RfPeqA8kLxuDK"
+                                       + "vePrj_7ntEap-s-2jcLVLf4W2MU6VaZFK4fs99daRtn6eHQWU0PwirIxa0miuzu3AenyMi43K9"
+                                       + "NemUma5lnIn_UU-_AxytfYoFBuVw5eKtmjEPW6H1foxjFqbM8ovKEhD7Iod7Rn1R6mQMhZpesk"
+                                       + "pQkutWvaViejNfM-lhO-eaQ");
+        httpHeaders.add(SERVICE_AUTHORIZATION, "eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJmcGxfY2FzZV9zZXJ2aWNlIiwi"
+                                               + "ZXhwIjoxNTkwNjg4NTU4fQ.I-Kuj5uEQqInvIn53bBoJtbtEYp5BGK-qakBjU5V_Mz"
+                                               + "ruwmGq9ctPo6kYhnSkzvQ81sasaHZXtjbSKWfnUNeZg");
+
+        objectMapper.registerModule(new Jdk8Module());
+    }
+
+    public PreSubmitCallbackResponseForTest aboutToSubmit(
+        CallbackForTestBuilder callback
+    ) throws Exception {
+
+        final String content = toJson(callback.build());
+
+        final MockHttpServletResponse response = mockMvc.perform(post(aboutToSubmitUrl)
+            .headers(httpHeaders)
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(content))
+            .andExpect(status().isOk()).andReturn().getResponse();
+
+        return translateException(() -> objectMapper
+            .readValue(response.getContentAsByteArray(), PreSubmitCallbackResponseForTest.class));
+    }
+
+    public PreSubmitCallbackResponseForTest aboutToStart(
+        CallbackForTestBuilder callback
+    ) throws Exception {
+
+        final MockHttpServletResponse response = mockMvc.perform(post(aboutToStartUrl)
+            .headers(httpHeaders)
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(toJson(callback.build())))
+            .andReturn().getResponse();
+
+        return translateException(() -> objectMapper
+            .readValue(response.getContentAsByteArray(), PreSubmitCallbackResponseForTest.class));
+    }
+
+    public PreSubmitCallbackResponseForTest ccdSubmitted(
+        CallbackForTestBuilder callback
+    ) throws Exception {
+
+        final MockHttpServletResponse response = mockMvc.perform(post(ccdSubmittedUrl)
+            .headers(httpHeaders)
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(toJson(callback.build())))
+            .andExpect(status().isOk()).andReturn().getResponse();
+
+        return translateException(() -> objectMapper
+            .readValue(response.getContentAsByteArray(), PreSubmitCallbackResponseForTest.class));
+    }
+
+    private String toJson(Object o) {
+        return translateException(() -> objectMapper.writeValueAsString(o));
+    }
+
+    private <T> T translateException(CallableWithException<T> callable) {
+        try {
+            return callable.call();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    interface CallableWithException<T> {
```

Comment:
can consider putting this in a separate file to make reusable.

Reply:
Done